kernel/aster-nix/src/vm/vmar/vm_mapping.rs

```diff
@@ -590,17 +595,19 @@ impl VmMappingInner {
         self.map_to_addr..self.map_to_addr + self.map_size
     }
 
-    fn check_perm(&self, page_idx: &usize, perm: &VmPerm) -> Result<()> {
-        // Check if the page is in current VmMapping.
-        if page_idx * PAGE_SIZE < self.vmo_offset
-            || (page_idx + 1) * PAGE_SIZE > self.vmo_offset + self.map_size
-        {
-            return_errno_with_message!(Errno::EINVAL, "invalid page idx");
-        }
+    fn check_perm(&self, perm: &VmPerm) -> Result<()> {
         if !self.perm.contains(*perm) {
             return_errno_with_message!(Errno::EACCES, "perm check fails");
         }
+        Ok(())
+    }
 
+    fn check_page_idx_range(&self, page_idx_range: &Range<usize>) -> Result<()> {
+        if page_idx_range.start * PAGE_SIZE < self.vmo_offset
+            || page_idx_range.end * PAGE_SIZE > self.vmo_offset + self.map_size
+        {
```

Comment:
Should `page_idx_range.end * PAGE_SIZE` be `(page_idx_range.end + 1) * PAGE_SIZE`?

Reply:
No, `page_idx_range.end` is the next page after the valid pages.